receiver/otlpreceiver/config.go

```diff
@@ -34,8 +36,14 @@ type Config struct {
 }
 
 var _ config.Receiver = (*Config)(nil)
+var _ config.Unmarshable = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid
 func (cfg *Config) Validate() error {
 	return nil
 }
+
+// Unmarshal a viper data into the config struct
+func (cfg *Config) Unmarshal(componentViperSection *viper.Viper) error {
+	return customUnmarshaler(componentViperSection, cfg)
```

Comment:
Same probably we should move the code here, so casting will not happen, nil check not needed, etc. :)